core/src/main/java/nu/yona/server/analysis/entities/WeekActivity.java

```diff
@@ -0,0 +1,59 @@
+package nu.yona.server.analysis.entities;
+
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+import nu.yona.server.entities.RepositoryProvider;
+import nu.yona.server.goals.entities.Goal;
+
+@Entity
+@Table(name = "WEEK_ACTIVITIES")
+public class WeekActivity extends IntervalActivity
+{
+	public static WeekActivityRepository getRepository()
+	{
+		return (WeekActivityRepository) RepositoryProvider.getRepository(WeekActivity.class, UUID.class);
+	}
+
+	@OneToMany(cascade = CascadeType.ALL)
+	private List<DayActivity> dayActivities;
+
+	// Default constructor is required for JPA
+	public WeekActivity()
+	{
+		super(null, null, null, null);
+	}
+
+	public WeekActivity(UUID id, UUID userAnonymizedID, UUID goalID, ZonedDateTime zonedStartOfWeek,
```

Comment:
This one can be private, right? It's only used from the ``createInstance`` method.